File: Examples/SecureRandom.java
import java.security.SecureRandom;

/* SecureRandom is Java's cryptographically-strong PRNG. The no-arg constructor
   picks the JVM's default secure source (NativePRNG on Unix-likes,
   SHA1PRNG/Windows-PRNG depending on configuration). For NIST-compliant DRBG
   selection use SecureRandom.getInstance("DRBG"). */
public class SecureRandom_64Bit {
  public static void main(String[] args) {
    SecureRandom rng = new SecureRandom();
    long randomNumber = rng.nextLong();
    System.out.printf("Random 64-bit number: %016x%n", randomNumber);
  }
}

File: Examples/SecureRandom.kt
// Kotlin (JVM): SecureRandom is the same as Java's java.security.SecureRandom,
// backed by the platform's OS entropy source. For Kotlin Multiplatform code
// you can use kotlin.random.Random.Default (non-cryptographic, fast) or
// kotlinx-crypto's SecureRandom for crypto-strong portability.

import java.nio.ByteBuffer
import java.security.SecureRandom

fun main() {
    val rng = SecureRandom()
    val bytes = ByteArray(8).also { rng.nextBytes(it) }
    val n = ByteBuffer.wrap(bytes).long
    println("Random 64-bit number: ${"%016x".format(n)}")
}
